haystack/nodes/ranker/recentness_ranker.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import warnings
+from collections import defaultdict
+from typing import List, Union, Optional, Dict, Literal
+
+from dateutil.parser import parse, ParserError
+
+from haystack.nodes.ranker.base import BaseRanker
+from haystack.schema import Document
+
+logger = logging.getLogger(__name__)
+
+
+class RecentnessRanker(BaseRanker):
+    outgoing_edges = 1
+
+    def __init__(
+        self,
+        date_identifier: str,
```

Comment:
I agree, date_meta_field is more consistent and more informative - implementing the change now, will ping you when all changes are implemented to take another look. 

As for setting the date param to default 'date', I'm not sure. I think if the date metadata field name is not provided by the user it should rather be flagged as mandatory param missing to emphasise the importance of adding this information. If it defaults to 'date' I can imagine a situation where you get an error "The metadata field 'date' was not found" and you need to remember - did you misname your metadata field (or made a small typo like naming it "Date" instead of "date")? Or did you forget to add metadata altogether? Or did you add the metadata under another field name but forgot to change it in the parameters? 

So, I think keeping it as a mandatory empty string to be filled in is a more straightforward approach, but of course if setting a default there would be more consistent with general Haystack user experience, I can change - let me know 👍

Reply:
Ah! This is supposed to be the name of field as is in the metadata, I completely misunderstood that.

I do agree that it must have no default but we need a better name for this, `date_meta_field` works for me.